src/components/buttons/download-button.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { useState } from 'react';
+
+import PropTypes from 'prop-types';
+
+import { useSnackbar } from 'notistack';
+import { useIntlRef } from '../../utils/messages';
+
+import { Button, CircularProgress } from '@mui/material';
+import { GetApp } from '@mui/icons-material';
+
+import { fetchFileFromProcess } from '../../utils/rest-api';
+
+const DownloadButton = ({ processFile, timestamp }) => {
+    const intlRef = useIntlRef();
+    const { enqueueSnackbar } = useSnackbar();
+
+    const [isLoading, setIsLoading] = useState(false);
+
+    const handleClick = () => {
+        setIsLoading(true);
+        downloadFile(processFile, timestamp, intlRef, enqueueSnackbar);
+    };
+
+    async function downloadFile(
```

Comment:
Why do you define the function inside the component and not outside as it was done before?

Reply:
To call this line

`setIsLoading(false);`

But indeed, i think it's better to extract the function and make my handleClick async to await downloadFile and set the isLoading boolean to false